src/Microsoft.FeatureManagement/FeatureFilters/Crontab/CrontabExpression.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+
+namespace Microsoft.FeatureManagement.FeatureFilters.Crontab
+{
+    /// <summary>
+    /// The Crontab expression with Minute, Hour, Day of month, Month, Day of week fields.
+    /// </summary>
+    public class CrontabExpression
+    {
+        private static readonly int NumberOfFields = 5;
+        private static readonly char[] WhitespaceDelimiters = {' ', '\n', '\t'};
+
+        private readonly CrontabField[] CrontabFields = new CrontabField[]
+        {
+            new CrontabField(CrontabFieldKind.Minute),
+            new CrontabField(CrontabFieldKind.Hour),
+            new CrontabField(CrontabFieldKind.DayOfMonth),
+            new CrontabField(CrontabFieldKind.Month),
+            new CrontabField(CrontabFieldKind.DayOfWeek)
+        };
+
+        /// <summary>
+        /// Check whether the given expression can be parsed by a Crontab expression.
+        /// If the expression is invalid, an ArgumentException will be thrown.
+        /// </summary>
+        /// <param name="expression">The expression to parse.</param>
+        /// <returns>A parsed CrontabExpression.</returns>
+        public static CrontabExpression Parse(string expression)
+        {
+            CrontabExpression crontabExpression = new CrontabExpression();
+            if (crontabExpression.TryParse(expression, out string message) == false)
```

Comment:
TryParse is a common pattern in .NET. Usually it contains an out parameter of what it parsed. This doesn't seem to fit.